From 1e89cbd7a8702bccc5166177724e12af55f21ac1 Mon Sep 17 00:00:00 2001
From: Alexander Koskovich <akoskovich@pm.me>
Date: Thu, 16 Apr 2026 22:41:07 +0000
Subject: [PATCH] net: ipa: fix IPA v5.5 configuration data

struct ipa_qmb_outstanding {
        u16 ot_reads;
        u16 ot_writes;
        u16 ot_read_beats;
};
[IPA_5_5][IPA_QMB_INSTANCE_DDR] = {16, 12, 0},
[IPA_5_5][IPA_QMB_INSTANCE_PCIE] = {16, 8, 0},

IPA_ENDPOINT_AP_LAN_RX:
        [IPA_5_5][IPA_CLIENT_APPS_LAN_CONS] = {
                        true, IPA_v5_5_GROUP_UL,
                        false,
                        IPA_DPS_HPS_SEQ_TYPE_INVALID,
                        QMB_MASTER_SELECT_DDR,
                        { 17, 14, 9, 9, IPA_EE_AP, GSI_ESCAPE_BUF_ONLY, 0       },
                        IPA_TX_INSTANCE_UL },

IPA_ENDPOINT_AP_MODEM_RX:
        [IPA_5_5][IPA_CLIENT_APPS_WAN_CONS] = {
                        true, IPA_v5_5_GROUP_DL,
                        false,
                        IPA_DPS_HPS_SEQ_TYPE_INVALID,
                        QMB_MASTER_SELECT_DDR,
                        { 24, 1, 9, 9, IPA_EE_AP, GSI_SMART_PRE_FETCH, 3        },
                        IPA_TX_INSTANCE_DL },

IPA_ENDPOINT_MODEM_AP_RX:
        [IPA_5_5][IPA_CLIENT_Q6_WAN_CONS]         = {
                        true, IPA_v5_5_GROUP_UL,
                        false,
                        IPA_DPS_HPS_SEQ_TYPE_INVALID,
                        QMB_MASTER_SELECT_DDR,
                        { 22, 7, 9, 9, IPA_EE_Q6, GSI_ESCAPE_BUF_ONLY, 0 },
                        IPA_TX_INSTANCE_UL },

Signed-off-by: Alexander Koskovich <akoskovich@pm.me>
[Imported from the RFC as posted to netdev; the archive hides the author's
 address, which is taken from his upstream commits. No code changes.]
Link: https://ratatoskr.run/netdev/2026/04/3516898
---
 drivers/net/ipa/data/ipa_data-v5.5.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/net/ipa/data/ipa_data-v5.5.c b/drivers/net/ipa/data/ipa_data-v5.5.c
index 741ae21d9d78..f6ba3b944700 100644
--- a/drivers/net/ipa/data/ipa_data-v5.5.c
+++ b/drivers/net/ipa/data/ipa_data-v5.5.c
@@ -50,13 +50,13 @@ enum ipa_rsrc_group_id {
 /* QSB configuration data for an SoC having IPA v5.5 */
 static const struct ipa_qsb_data ipa_qsb_data[] = {
 	[IPA_QSB_MASTER_DDR] = {
-		.max_writes		= 0,	/* Unlimited */
-		.max_reads		= 12,
+		.max_writes		= 12,
+		.max_reads		= 0,	/* Unlimited */
 		.max_reads_beats	= 0,
 	},
 	[IPA_QSB_MASTER_PCIE] = {
-		.max_writes		= 0,	/* Unlimited */
-		.max_reads		= 8,
+		.max_writes		= 8,
+		.max_reads		= 0,	/* Unlimited */
 		.max_reads_beats	= 0,
 	},
 };
@@ -86,8 +86,8 @@ static const struct ipa_gsi_endpoint_data ipa_gsi_endpoint_data[] = {
 	},
 	[IPA_ENDPOINT_AP_LAN_RX] = {
 		.ee_id		= GSI_EE_AP,
-		.channel_id	= 13,
-		.endpoint_id	= 16,
+		.channel_id	= 14,
+		.endpoint_id	= 17,
 		.toward_ipa	= false,
 		.channel = {
 			.tre_count	= 256,
@@ -135,7 +135,7 @@ static const struct ipa_gsi_endpoint_data ipa_gsi_endpoint_data[] = {
 	[IPA_ENDPOINT_AP_MODEM_RX] = {
 		.ee_id		= GSI_EE_AP,
 		.channel_id	= 1,
-		.endpoint_id	= 23,
+		.endpoint_id	= 24,
 		.toward_ipa	= false,
 		.channel = {
 			.tre_count	= 256,
@@ -168,7 +168,7 @@ static const struct ipa_gsi_endpoint_data ipa_gsi_endpoint_data[] = {
 	[IPA_ENDPOINT_MODEM_AP_RX] = {
 		.ee_id		= GSI_EE_MODEM,
 		.channel_id	= 7,
-		.endpoint_id	= 21,
+		.endpoint_id	= 22,
 		.toward_ipa	= false,
 	},
 	[IPA_ENDPOINT_MODEM_DL_NLO_TX] = {
